components/constellation/pipeline.rs

```diff
@@ -624,8 +609,10 @@ impl UnprivilegedPipelineContent {
         );
 
         if wait_for_completion {
-            let _ = self.script_content_process_shutdown_port.recv();
-            let _ = self.layout_content_process_shutdown_port.recv();
+            let signal = content_process_shutdown_port.recv();
+            // When the script-thread drops the sender,
+            // we expect to receive an Err(_).
+            assert!(signal.is_err());
```

Comment:
I'm confused about who sends on this channel, since the only uses of `send(())` appear to have been deleted. Are we using a channel just for its `Drop` implementation?

Reply:
Yes, previously there was an explicit sending of a message when the script-thread exits, which I can bring back if think that's clearer. Or we can effectively wait until the channel drops, and assert it is indeed the `Err` that would be received in such a case. 

I think we can count on the channel not sending any spurious errors.

Reply:
Actually, I think you're right it's better to send an explicit message, since the drop could also be the result of an unexpected shutdown of the script-thread, while a message will always be explicitly sent as part of graceful shutdown...